Develop alim

alimuzzaman opened the pull request.

Commit:
Fixed mimeType #371

Commit:
Update class-bootstrap.php

Multisite wrong GCS path
#407

Commit:
wildcards for bucket folder settings #149

Commit:
wildcards for bucket folder settings #149

Commit:
Change default value for CACHE BUSTING #361

Commit:
Metadata for attachment missing with Polylang Pro #378

Commit:
Update class-bootstrap.php

Hide Regenerate and Sync with GCS when mode is Disabled. #440

Commit:
New endpoint for the Google Cloud Storage JSON API #384

Commit:
wildcards for bucket folder settings #149

Commit:
wildcards for bucket folder settings #149

Commit:
wildcards for bucket folder settings #149

Commit:
Issue #149

planv committed:
fix conflicts

planv committed:
#149 added wildcards to existing sync methods

planv committed:
#149 updated class upgrader

Commit:
Merge branch 'latest' into develop-alim

Commit:
Remove the root dir only from the beginning

#149

Commit:
Merge branch 'develop-vova' into develop-alim

planv committed:
#149 added wildcards to existing sync methods

Commit:
Fix for Undefined index: path

#378

Commit:
Merge branch 'develop-vova' into develop-alim

Commit:
Update class-utility.php

#378

Commit:
Multisite fixes #149

Commit:
Merge branch 'multisite' into develop-alim

Commit:
Added default value for root dir and improved upgrade process

#149

Commit:
Code improvement  #149

Commit:
Update class-bootstrap.php

#461

Commit:
Fix for Undefined index: use_wildcards

Commit:
Update class-sm-cli-sync.php

#451

Commit:
Update class-upgrader.php

#149

Commit:
Allow --force param to be passed on each batch command

#451

Commit:
Update class-sm-cli-command.php

#451

planv committed:
#378 Metadata for attachment missing with Polylang Pro. Fixed

planv committed:
rename and relocate bucket folder setting #463

planv committed:
rename and relocate bucket folder setting #463

Commit:
Merge branch 'develop-vova' into develop-alim

Commit:
Update ICompatibility.php

Issue #464

Commit:
wildcards for bucket folder settings #149

Commit:
Initialize settings object in bootstrap construct

Commit:
Fix for strpos(): Empty needle #149

`

Commit:
wildcards for bucket folder settings #149

Commit:
Fix for undefined function wp_verify_nonce()

Commit:
Update class-settings.php

Initialize settings object in bootstrap construct #465

Commit:
wildcards for bucket folder settings #149

#461

Commit:
Initialize settings object in bootstrap construct #465

planv committed:
#449 added new option for CLI

Commit:
Update class-bootstrap.php

wildcards for non media file #149

Commit:
Version bump 2.4.0-RC.1

Commit:
Issue with network override of Cache-Busting #468

Commit:
WP Stateless Elementor - CSS files wrong MIME type #395

Commit:
Create db.json

WP Stateless Elementor - CSS files wrong MIME type #395

Commit:
Revert "Create db.json"

This reverts commit 29fac70452a96f5ee12530ac92dd02317675e4a7.

Commit:
Revert "WP Stateless Elementor - CSS files wrong MIME type #395"

This reverts commit d5758c68918d1c13f9be08d074956d2c8a1587aa.

Commit:
WP Stateless Elementor - CSS files wrong MIME type #395

Commit:
Don't use wildcard for non-media.

Issue #149

Commit:
Don't use wildcard for non-media

Issue #149

Commit:
Use sites/id prefix for non-media files

Issue #149

planv committed:
#149 fixed removing non-media files

Commit:
Image optimization compatibility

Issue #149

planv committed:
#149 wildards for multisite

planv committed:
#461 wildcards for multisite

planv committed:
#461 wildcards for multisite

planv committed:
#461 wildcards for mainsite

planv committed:
#149 fixed compatibility for ewww plugin

planv committed:
#149 fixed compatibility for ewww plugin

planv committed:
#149 fixed compatibility for retina-2x plugin

planv committed:
#149 sync images first time without using wildcards

Commit:
Review migration #462

Commit:
Multisite fix #462

Commit:
version bump RC.3

antonkorotkov commented on lib/classes/class-gs-client.php:
```diff
@@ -189,12 +190,17 @@ public function add_media( $args = array() ) {
             return new \WP_Error( 'sm_error', __( 'Unable to locate file on disk', ud_get_stateless_media()->domain ) );
           }
 
+          $use_wildcards = isset($_REQUEST['use_wildcards']) ? $_REQUEST['use_wildcards'] : false;
```
This string appears as a duplicate many times. Probably it is better to create some utility function that will return whether to use_wildcards or not.

planv replied:
Done

planv committed:
#149 utility fixes

planv committed:
#149 removed fix for current images

Commit:
Update lite-speed-cache.php

Commit:
Update class-utility.php

Fix for new upload. Issue #149

Commit:
Update class-settings.php

Refresh setting after saving to DB.

planv committed:
#462 added fix for migration from version 2.1

Commit:
Merge branch 'develop-vova' into develop-alim

planv committed:
#462 wildcards fix

Commit:
Update woo-extra-product-options.php

Woocommerce extra product options #266
#149

planv committed:
Merge remote-tracking branch 'origin/develop-alim' into develop-vova

Commit:
Update woo-extra-product-options.php

Commit:
Exclude sites/%site_id% when upgrade

issue #462

Commit:
Update wp-stateless.js

Fixed preview URL when root dir is empty

planv committed:
#462 added plugin version to sm_cloud meta

planv committed:
#149 changed wildcards logic

planv committed:
#149 fixed using wildcards for function remove_media

planv committed:
#149 forcing  for multisites

planv committed:
#149 fixed adding new files on sync process via dynamic-image-support

planv committed:
#149 fixed compatibility for multisite

planv committed:
Code refactoring

planv committed:
#149 fixed not existing wildcards

planv committed:
#149 fixed compatibility for multisite

planv committed:
#149 fixed not existing wildcards

planv committed:
#149 fixed not existing wildcards

Commit:
fix gs name for thumbnail #149

planv committed:
#446 updated Client library for Google APIs

planv committed:
#149 fix for removing .webp on GCS

planv committed:
#149 checked todo

planv committed:
#149 added underscore to regexp for Folder value

planv committed:
#383 Optimize and minimize sm_cloud meta, removed not used fields

planv committed:
added compatibility with Imagify version 1.9, also supporting webp format issue #403

planv committed:
record generated thumbnail sizes for each attachment posts #268

planv committed:
#470 added metabox.io

planv committed:
recreated attachment metabox panel using metabox.io #470

planv committed:
#470 added UI fixes, function for select all rregistered sized

planv committed:
#149 fix for count stateless media function

planv committed:
#471 added link to attachment detail page instead metabox

planv committed:
#471 added link to attachment detail page instead metabox

Commit:
#470 moved metabox tab to lib dir

planv committed:
#470 changes on metabox, added sync for selected size

planv committed:
#473 added button to sync new media

planv committed:
#392 fix compatibility with PHP since version 7

Commit:
Update regenerate_interface.php

Allow sync mode to be added from extension.

Commit:
#478 base64_encode memory limit

Commit:
#403 Fix for stateless mode

planv committed:
#149 added dot to regexp for Folder value

Commit:
#403 Imagify compatability

planv committed:
nerge conflicts

planv committed:
#481 rename current Stateless mode to Ephemeral

planv committed:
metabox fix

planv committed:
added dependencies for google/cloud-core and google/cloud-storage

Commit:
#478 base64_encode memory limit

planv committed:
#403 retrieved attachment ID from media object

planv committed:
#482 add new stateless mode

planv committed:
Merge remote-tracking branch 'origin/develop-alim' into develop-vova

planv committed:
#482 new mode stateless. fixes for media files which was uploaded on local server

planv committed:
#482 sync old media ffiles with new sateless mode

planv committed:
#482 added hooks for new sateless mode, forcing Cache-Busting

planv committed:
#482 added GCS metadata for new mode

planv committed:
#482 added GCS metadata for new mode

planv committed:
#482 sync process, removing media from local server

planv committed:
added compatibility for Google App Engine #486

planv committed:
compatibility for Google App Engine #486

planv committed:
compatibility for Google App Engine #486, showing thumbnails

Commit:
Update class-upgrader.php

rename current "Stateless" mode to "Ephemeral"
#481

planv committed:
Merge remote-tracking branch 'origin/develop-alim' into develop-vova

planv committed:
#149 forcing Cache-Busting only or multisites

planv committed:
fix for metabox display on new post page

planv committed:
#482 added predefined values to field Folder on Stateless mode

planv committed:
#482 set Stateless mode by default if folder Uploads is not writable

Commit:
#403 Imagify compatability

planv committed:
#486 new mode for non-media files

planv committed:
Merge remote-tracking branch 'origin/develop-alim' into develop-vova

planv committed:
#482 use our Folder settings if it is set, instead default

planv committed:
#482 added wildcards for new Stateless mode for non-media-files

planv committed:
#482 disabled compatibilities which not working on Stateless mode

planv committed:
#486 added fixes for CLI functionality

Commit:
Update lite-speed-cache.php

planv committed:
#491 fixes for new version of Litespeed Cache

Commit:
Update class-utility.php

compatibility for Google App Engine #486

planv committed:
#486 fixed variable name

Commit:
#486 upload to root dir in Google App Engine

planv committed:
#486 fixes for multisite

planv committed:
Merge remote-tracking branch 'origin/develop-alim' into develop-vova

planv committed:
#486 added fixes for new mode

planv committed:
#493 convert wildcards to real data

planv committed:
#493 using year and month from current path instead get from current time

planv committed:
#482 fix fir srcset on Stateless mode

planv committed:
#482 fix for the_content_filter and removing double year/month from path

planv committed:
#486 clear subdir for Stateless mode

Commit:
#493 removed year month wildcard

Commit:
#493 removed year month wildcard

Commit:
#491 fix for Litespeed fatal error

Commit:
#491 fix for copy media

Commit:
#491 fix for mimetype

Commit:
Update class-utility.php

#491 fix for mimetype

planv committed:
#482 prevent google storage init if json key is empty

Commit:
wp_stateless_file_name filter #493

Commit:
#493

Commit:
#493 Revert using year and month from current path instead get from current …

Commit:
Fixed ShortPixel webP issue

Commit:
#493 updated migration for Bucket Folder.

Commit:
Merge branch 'develop-vova' into develop-alim

Commit:
#486 catching fatal error.

$object->update generate fatal error if image upload was faild in previous step.

Commit:
#501 add compatibility for Gravity Forms Signature Add-On

Commit:
Revert " #501 add compatibility for Gravity Forms Signature Add-On"

This reverts commit 277dde1a832d7605feafcc058045d908142999cb.

Commit:
#501 compatibility for Gravity Forms Signature Add-On

planv committed:
#502 sort all compatibilities alphabetically

planv committed:
#502 sort all compatibilities alphabetically

Commit:
Merge branch 'develop-vova' into develop-alim
